aws-android-sdk-pinpoint/src/main/java/com/amazonaws/mobileconnectors/pinpoint/internal/event/EventRecorder.java

```diff
@@ -436,16 +437,27 @@ private void submitEventsAndEndpoint(final JSONArray eventArray,
     }
 
     private void processEndpointResponse(EndpointProfile endpoint, PutEventsResult resultResponse) {
-        final EndpointItemResponse endpointItemResponse = resultResponse
+        final Map<String, ItemResponse> results = resultResponse
                 .getEventsResponse()
-                .getResults()
-                .get(endpoint.getEndpointId())
-                .getEndpointItemResponse();
-        if(202 == endpointItemResponse.getStatusCode()) {
-            log.info("EndpointProfile updated successfully.");
+                .getResults();
+        if (results.isEmpty()) {
```

Comment:
Looks like we are going to `if else` arrow here. Whenever a control does not need to continue after an `if` block, we can "short-circuit" the block by returning from there. This lets reader know that rest of the code is not executed and saves many tabs as bonus 😸